R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,702 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "user_name = readline('Username: ')\n",
+    "user_password <- getPass::getPass('OMERO password: ')\n",
+    "server <- OMEROServer(host = 'outreach.openmicroscopy.org', username=user_name, password=user_password, port= as.integer(4064))\n",
+    "server <- connect(server)\n",
+    "paste('Successfully logged in as', server@user$getUserName())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load and display an image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load an image from the example dataset 'R-dataset' (these images are part of the original 'CDK5RAP2-C' dataset).\n",
+    "\n",
+    "**Exercise:** Go to the image 'siControl_N20_Cep215_I_20110411_Mon-1509_0_SIR_PRJ.dv' in your 'R-dataset', find the 'Image ID' and copy/paste it below. You can use a different image of the dataset if you like, but I recommend that one because it nicely shows the toroid centrioles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imageId <- REPLACE_WITH_IMAGE_ID # An image from R-dataset\n",
+    "image <- loadObject(server, \"ImageData\", imageId)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load the pixel values of the second channel (the CDK5RAP2-C channel) and display the image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pixels <- getPixelValues(image, 1, 1, 2) # Second channel of the first z plane of the first time point\n",
+    "                                         # (the images only have one z/t plane anyway)\n",
+    "ebi <- EBImage::Image(data = pixels, colormode = 'Grayscale')\n",
+    "ebi <- normalize(ebi)\n",
+    "EBImage::display(ebi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Image segmentation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Visual"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "threshImg <- thresh(ebi, w=15, h=15, offset=0.1)\n",
+    "threshImg <- medianFilter(threshImg, size=3)\n",
+    "threshImg <- fillHull(threshImg)          \n",
+    "threshImg <- bwlabel(threshImg)\n",
+    "EBImage::display(colorLabels(threshImg))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Modify the parameters of the thresh function to get a feeling for how the parameters affect the segmentation, or even plug in other methods into the segmentation workflow to improve the result, see [EBImage documentation](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2)"
```

Comment:
``....or even plug in...`` Leave this part of the sentence out. If you want to, declare another (more advanced) exercise instead. Going to the unfamiliar doc and looking for an entity which I do not fully understand is suddenly quite a jump.

Reply:
After I have executed this cell, I am getting a very unpleasant effect - not seeing the resulting image without scrolling.

![screen shot 2018-11-21 at 12 09 03](https://user-images.githubusercontent.com/2478303/48840378-5870fb00-ed86-11e8-8565-46fd9ec998ee.png)

Reply:
It is because of the width and height parameters - this is not explained in the exercise. The starting parameters for w and h should be something like 
``threshImg <- thresh(ebi, w=15, h=40, offset=0.1)``
which for me displays the whole image at least.

Reply:
This exercise is simply confusing. 

When I hear ``thresh`` I immediately think about ``thresholding``. The problem is that in the first line, I am actually manipulating the size of the output -not at all the usual thresholding stuff. The function is not explained. Neither it is intuitive. And the link to the doc is far too general (send me directly to the explanation of the ``thresh`` function please.

Reply:
It is the ``offset`` parameter which really does something more intuitive. But, alas, it has not a very intuitive name...

Also, you did not explain what the objects to be thresholded for in the image are...

Reply:
You need to offer this link https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/thresh - and quite soon after the start of the cell/exercise.

Reply:
w, h and offset are thresholding parameters, nothing to do with the image size of the output. That the image size changed, must have been some weird random artifact. But yes, a link to the documentation of the function is definitely needed earlier.

Reply:
w and h are "width and height of the moving rectangular window". Which is not very intuitive. In fact still have no idea what is it doing. Yes, you are right, it is not resizing the output window. But neither it is intuitive, nor it intuitively belongs to a thresholding function.